Client/game_sa/CBuildingsPoolSA.cpp

```diff
@@ -98,6 +95,22 @@ CBuilding* CBuildingsPoolSA::AddBuilding(CClientBuilding* pClientBuilding, uint1
 
     // Add building in world
     auto pBuildingSA = new CBuildingSA(pBuilding);
+
+    if (pBuilding->HasMatrix())
+    {
+        // Edge case for the traincross2 (1374) model
+        // LoadObjectInstance allocates a matrix for the model
+        // We need allocate our own matrix and put the old matrix in the original pool
+        pBuildingSA->ReallocateMatrix();
+    }
+    else if (vRot->fX != 0 || vRot->fY != 0)
+    {
+        // Allocate matrix in our unlimited storage instead of using the shared pool.
+        pBuildingSA->AllocateMatrix();
+    }
```

Comment:
I would've just deallocated the game's matrix if it exists, and then only allocate it when `if (vRot->fX != 0 || vRot->fY != 0)`

Reply:
The game expects an allocated matrix for this specific model in 0x5A0B50 (CObject::ProcessTrainCrossingBehaviour)

Reply:
Ok, then it's fine. You responded with a `No` to such a question before, but maybe you discovered this after that.